packages/jellyfish-transaction-builder/__tests__/txn/txn_builder_set_default_loan_scheme.test.ts

```diff
@@ -0,0 +1,150 @@
+      identifier: ''
+    }, script)
+
+    const promise = sendTransaction(testing.container, txn)
+    await expect(promise).rejects.toThrow(DeFiDRpcError)
+    await expect(promise).rejects.toThrow('DefaultLoanSchemeTx: id cannot be empty or more than 8 chars long (code 16)\', code: -26')
+  })
+
+  it('should not setDefaultLoanScheme if identifier is more than 8 chars long', async () => {
+    const script = await providers.elliptic.script()
+    const txn = await builder.loans.setDefaultLoanScheme({
+      identifier: '123456789'
+    }, script)
+
+    const promise = sendTransaction(testing.container, txn)
+    await expect(promise).rejects.toThrow(DeFiDRpcError)
+    await expect(promise).rejects.toThrow('DefaultLoanSchemeTx: id cannot be empty or more than 8 chars long (code 16)\', code: -26')
+  })
+
+  it('should not setDefaultLoanScheme if identifier does not exists', async () => {
+    const script = await providers.elliptic.script()
+    const txn = await builder.loans.setDefaultLoanScheme({
+      identifier: 'scheme3'
+    }, script)
+
+    const promise = sendTransaction(testing.container, txn)
+    await expect(promise).rejects.toThrow(DeFiDRpcError)
+    await expect(promise).rejects.toThrow('DefaultLoanSchemeTx: Cannot find existing loan scheme with id scheme3 (code 16)\', code: -26')
+  })
+
+  it('should not setDefaultLoanScheme if the identifier is a default scheme', async () => {
+    const script = await providers.elliptic.script()
+    const txn = await builder.loans.setDefaultLoanScheme({
+      identifier: 'scheme1'
+    }, script)
+
+    const promise = sendTransaction(testing.container, txn)
+    await expect(promise).rejects.toThrow(DeFiDRpcError)
+    await expect(promise).rejects.toThrow('DefaultLoanSchemeTx: Loan scheme with id scheme1 is already set as default (code 16)\', code: -26')
+  })
+
+  it('should not setDefaultLoanScheme if the scheme is going to be deleted at future block', async () => {
+    await testing.container.call('createloanscheme', [300, new BigNumber(3.5), 'scheme3'])
+    await testing.generate(1)
+
+    // Wait for block 110
+    await testing.container.waitForBlockHeight(110)
+
+    // To delete at block 120
+    const loanSchemeId = await testing.container.call('destroyloanscheme', ['scheme3', 120])
+    expect(typeof loanSchemeId).toStrictEqual('string')
+    expect(loanSchemeId.length).toStrictEqual(64)
+    await testing.generate(1)
+
+    const script = await providers.elliptic.script()
+    const txn = await builder.loans.setDefaultLoanScheme({
+      identifier: 'scheme3'
+    }, script)
+
+    const promise = sendTransaction(container, txn)
+    await expect(promise).rejects.toThrow(DeFiDRpcError)
+    await expect(promise).rejects.toThrow('DefaultLoanSchemeTx: Cannot set scheme3 as default, set to destroyed on block 120 (code 16)\', code: -26')
+
+    // Delete at block 120
+    await testing.container.waitForBlockHeight(120)
```

Comment:
minor. probably could remove this since the purpose of the test is to test `should not setDefaultLoanScheme if the scheme is going to be deleted at future block`

Reply:
Ya it seems pointless to have the code as this is the last line of this test file.
But consistent wise, I have waitForBlockHeight at the beginning of `it` and end of `it` for all the code